File: app/src/main/java/com/thebook/bottomnav/ui/info/InfoViewModel.java
package com.thebook.bottomnav.ui.info;

import android.content.Context;
import android.content.SharedPreferences;
import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

/**
 * InfoViewModel - ViewModel for the Info screen
 * 
 * This ViewModel manages data for the InfoFragment, which displays detailed
 * information about selected movies. Currently it provides basic text content
 * stored in SharedPreferences, but could be extended to handle specific
 * movie detail data.
 */
public class InfoViewModel extends AndroidViewModel {
    // LiveData for holding text content to display
    private MutableLiveData<String> mText;
    
    // SharedPreferences key (currently unused but kept for potential future use)
    private static String MY_PREFS_NAME = "filename";

    /**
     * Constructor - Initializes the ViewModel and sets up text content
     * @param application Application context for accessing SharedPreferences
     */
    public InfoViewModel(@NonNull Application application) {
        super(application);
        
        // Access SharedPreferences for storing/retrieving data
        SharedPreferences prefs = getApplication()
          .getSharedPreferences("movie", Context.MODE_PRIVATE);

        // Initialize LiveData
        mText = new MutableLiveData<>();

        // Store a description in SharedPreferences
        // (This demonstrates SharedPreferences usage but could be improved)
        SharedPreferences.Editor preferencesEditor = prefs.edit();
        preferencesEditor
          .putString("description", "This is a dash board fragment");
        preferencesEditor.apply();

        // Retrieve the description and set it as the text value
        String description = prefs.getString("description", "");
        mText.setValue(description);
    }

    /**
     * Provides access to the text LiveData for UI components to observe
     * @return LiveData containing text content for display
     */
    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/thebook/bottomnav/ui/info/InfoViewModel.kt
package com.thebook.bottomnav.ui.info

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

/**
 * InfoViewModel - ViewModel for the Info screen
 * 
 * Converted to Kotlin from Java. This ViewModel manages data for the InfoFragment
 * which displays detailed information about selected movies.
 */
class InfoViewModel(application: Application) : AndroidViewModel(application) {

    companion object {
        private const val PREFS_NAME = "movie"
    }

    private val _text = MutableLiveData<String>()
    val text: LiveData<String> = _text

    init {
        val prefs = getApplication<Application>().getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        
        // Store description in SharedPreferences
        prefs.edit().apply {
            putString("description", "This is an info fragment")
            apply()
        }

        // Retrieve and set the description
        val description = prefs.getString("description", "")
        _text.value = description
    }
}
